src/MassTransit.RabbitMqTransport/Topology/RabbitMqExchangeBindingExtensions.cs

```diff
@@ -63,9 +67,9 @@ public static ExchangeBindingSettings GetExchangeBinding(this SendSettings setti
 
         public static IEnumerable<ExchangeBindingSettings> GetExchangeBindings(this ReceiveSettings settings, string exchangeName)
         {
-            var exchange = new Exchange(exchangeName, settings.Durable, settings.AutoDelete, settings.ExchangeType);
-
-            var binding = new ExchangeBinding(exchange);
+            var exchange = new Exchange(exchangeName, settings.Durable, settings.AutoDelete, settings.ExchangeTypeProvider);
+            //TODO find a way for this to stay working while you want to implement routingkey structure

```

Comment:
I found out that the logic in this method is directly related to the fanout exchange binding to the receiving queue.
Therefore it will not conflict with a custom routing structure.
That's why I removed the TODO and the commented code.